server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -1230,6 +1242,142 @@ public void testRunConcurrentSnapshots() {
         }
     }
 
+    public void testSnapshotCompletedByNodeLeft() {
```

Comment:
Rather than adding a test that reliably reproduces the failure in https://github.com/elastic/elasticsearch/issues/104939 I've elected to write this more general test with enough randomisation to cover a variety of similar situations too. It does reproduce the failure in https://github.com/elastic/elasticsearch/issues/104939 much more frequently than `SnapshotStressTestsIT`, but even so it might take a few hundred iterations to hit the problem.